### PR TITLE
[Fix] Return None instead of error if table info is not available on the node.

### PR DESCRIPTION
**Pull request by grao1991**
## Description
We don't have table info on non-indexer FN. When implementing the internal indexer for storage sharding, we accidentally change the logic to return an Error rather than Ok(None) when internal indexer is enabled but the table info indexer is not enabled. The PR fix the behavior to return Ok(None).

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [x] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by larry-aptos** [comment]
Looks good for cherry-pick; we should do a follow-up that cleans up the `indexer_enabled`. Currently, the meaning is confusing since table_info and internal index db are separate now.

**Review by larry-aptos — APPROVED** [review]

**Review by sherry-x — APPROVED** [review]

File: api/types/src/convert.rs [hunk 1]
```diff
@@ -1011,14 +1011,9 @@ impl<'a, S: StateView> MoveConverter<'a, S> {
...
     fn get_table_info(&self, handle: TableHandle) -> Result<Option<TableInfo>> {
         if let Some(indexer_reader) = self.indexer_reader.as_ref() {
-            // Attempt to get table_info from the indexer_reader if it exists
-            Ok(indexer_reader.get_table_info(handle)?)
-        } else if self.db.indexer_enabled() {
-            // Attempt to get table_info from the db if indexer is enabled
-            Ok(Some(self.db.get_table_info(handle)?))
-        } else {
-            Ok(None)
+            return Ok(indexer_reader.get_table_info(handle).unwrap_or(None));
```
Comment (larry-aptos) [inline api/types/src/convert.rs:1014]: does this mean if rare error happens here, we'll get nothing. we should be more explicit on the errors.

**Comment by sherry-x** [comment]
Please update the PR with better description

File: api/types/src/convert.rs (hunk 1 above)
Comment (grao1991) [reply api/types/src/convert.rs:1014]: Well the current implementation of get_table_info never returns error, so this PR doesn't make it worse.. It's not ideal, we can make improvement separately..

**Pull request merged**